packages/mgt-components/src/components/mgt-todo/mgt-todo.ts

```diff
@@ -266,66 +253,69 @@ export class MgtTodo extends MgtTasksBase {
-      </fluent-button>
+      <input
+        type="date"
+        id="new-taskDate-input"
+        class="date"
+        label="new-taskDate-input"
+        aria-label="new-taskDate-input"
+        role="textbox"
+        .value="${this.dateToInputValue(this._newTaskDueDate)}"
+        @change="${(e: Event) => {
+          const value = (e.target as HTMLInputElement).value;
+          if (value) {
+            this._newTaskDueDate = new Date(value + 'T17:00');
+          } else {
+            this._newTaskDueDate = null;
+          }
+        }}"
+      />
     `;
 
     const taskTitle = html`
-       <fluent-text-field
-         appearance="outline"
-         label="new-taskName-input"
-         aria-label="new-taskName-input"
-         .value=${this._newTaskName}
-         placeholder="${this.strings.newTaskPlaceholder}"
-         @keypress="${(e: KeyboardEvent) => {
-           if (e.key === 'Enter') {
-             this.addTask();
-           }
-         }}"
-         @input="${(e: Event) => {
-           this._newTaskName = (e.target as HTMLInputElement).value;
-           this.requestUpdate();
-         }}">
-          <div slot="start">${addIcon}</div>
-          <div slot="end">
-          ${calendarTemplate}
-          ${cancelIcon}</div>
-        </fluent-text-field>
-     `;
+      <fluent-text-field
+        appearance="outline"
+        id="new-taskName-input"
+        label="new-taskName-input"
+        aria-label="new-taskName-input"
+        .value=${this._newTaskName}
+        placeholder="${this.strings.newTaskPlaceholder}"
+        @keydown="${(e: KeyboardEvent) => {
+          if (e.key === 'Enter') {
+            this.addTask();
+          }
+        }}"
+        @input="${(e: Event) => {
+          if ((e.target as HTMLInputElement).id === 'new-taskName-input') {
+            this._newTaskName = (e.target as HTMLInputElement).value;
+            this.requestUpdate();
+          }
+        }}">
+        <div slot="start">${addIcon}</div>
+        <div slot="end">
+        <span class="calendar">${calendarTemplate}</span>
+        ${cancelIcon}</div>
+      </fluent-text-field>
+    `;
     return html`
-      <div dir=${this.direction} class="Task NewTask Incomplete">
-        ${taskTitle}
-      </div>
-        ${
-          this._openCalendar
-            ? html`
-                <fluent-calendar readonly="false" locale="en-US"
-                  selected-dates="${this._newTaskDueDate}"
-                  @dateselected="${e => this.handleDateSelected(e)}"
-                  @mouseleave=${() => this.toggleCalendar(false)}>
-                </fluent-calendar>`
-            : null
-        }
+      ${
+        this.currentList
+          ? html`
+            <div dir=${this.direction} class="Task NewTask Incomplete">
+              ${taskTitle}
+            </div>
+        `
+          : html``
+      }  
      `;
   }
-  // selected-dates="2-20-2023"
+
   protected async handleSelectionChanged(e: any) {
```

Comment:
We shoule have a type on the event here please, it's a CustomEvent<T> where T is defined by the code emitting the event